language-server/tests/dm_tests.ml

```diff
@@ -156,11 +156,55 @@ let%test_unit "step_forward.delete_observe_id" =
   let st, events = DocumentManager.interpret_to_previous st in
   let st = handle_events todo st in
   [%test_eq: bool] (Option.is_none (DocumentManager.Internal.observe_id st)) true
-  
+
+  (* With this test we can check that interpret_in_background has lower priority then interpret to *)
+  let%test_unit "interpret_in_background.interpret_to stateful" = 
+  let st, init_events = init "Definition x := true. Definition y := false. Definition z := 0." in
+  let st, (s1, (s2, (s3, ()))) = dm_parse st (P (P (P O))) in
+  let todo = Sel.(enqueue empty init_events) in
+  let st = handle_events todo st in 
+  let st1, events = DocumentManager.interpret_in_background st in
+  let todo = Sel.(enqueue empty events) in 
+  let position = RawDocument.position_of_loc (DocumentManager.Internal.raw_document st) s2.stop in
+  let st2, events = DocumentManager.interpret_to_position ~stateful:true st1 position in 
+  let todo = Sel.(enqueue todo events) in 
+  let st = handle_events todo st1 in 
+  [%test_pred: sentence_id option] (Option.equal Stateid.equal (Some s3.id)) (DocumentManager.Internal.observe_id st)
+
+  (* With this test interpret_to_end and interpret_to have the same priority, and interpret to is stateful 
+     so it will modify observe id, they will get executed in order of insertion, hence observe_id = s2.id *)  
+  let%test_unit "interpret_to_end.interpret_to stateful" = 
+  let st, init_events = init "Definition x := true. Definition y := false. Definition z := 0." in
+  let st, (s1, (s2, (s3, ()))) = dm_parse st (P (P (P O))) in
+  let todo = Sel.(enqueue empty init_events) in
+  let st = handle_events todo st in 
+  let st, events = DocumentManager.interpret_to_end st in
+  let todo = Sel.(enqueue empty events) in 
+  let position = RawDocument.position_of_loc (DocumentManager.Internal.raw_document st) s2.stop in
+  let st, events = DocumentManager.interpret_to_position ~stateful:true st position in 
+  let todo = Sel.(enqueue todo events) in 
+  let st = handle_events todo st in 
+  [%test_pred: sentence_id option] (Option.equal Stateid.equal (Some s2.id)) (DocumentManager.Internal.observe_id st)
+
+  (* With this test interpret_to_end and interpret_to have the same priority, and interpret to is not stateful 
+     so it will not modify observe id, they will get executed in order of insertion, hence observe_id = s3.id *) 
+  let%test_unit "interpret_to_end.interpret_to not stateful" = 
+  let st, init_events = init "Definition x := true. Definition y := false. Definition z := 0." in
+  let st, (s1, (s2, (s3, ()))) = dm_parse st (P (P (P O))) in
+  let todo = Sel.(enqueue empty init_events) in
+  let st = handle_events todo st in 
+  let st, events = DocumentManager.interpret_to_end st in
+  let todo = Sel.(enqueue empty events) in 
+  let position = RawDocument.position_of_loc (DocumentManager.Internal.raw_document st) s2.stop in
+  let st, events = DocumentManager.interpret_to_position ~stateful:false st position in 
```

Comment:
Is `stateful:false` used only in tests?

Reply:
Nope ! It is used in the [lspManager](https://github.com/coq-community/vscoq/blob/a9ca17fa9cc9a4dba51a3f716edbb45150cfb565/language-server/vscoqtop/lspManager.ml#LL255C42-L255C42) to launch higher priority "interpret_to" commands that are not stateful in continuous mode. This is necessary to update the goal view.